+replab/Centralvar.m

```diff
@@ -0,0 +1,232 @@
+classdef Centralvar < replab.Str
+% A matrix variable satisfying some symmetry constraints
+%
+% example: replab.Centralvar.fromGenerators({[3 1 2]})
+
+% Current limitations:
+% - The sdp matrix produced is currently always square, real, and symmetric
+% - Only supports complex and quaternionic representations with dimension 2
+%   and 4 respectively.
+% - limited arithmetic, essentially we can just:
+%   - access the elements of the matrix through the fullMatrix method
+%   - define an SDP constraint with a constant bound, such as "M >= 0"
+
+    properties (SetAccess = protected)
+        U; % Unitary operator block-diagonalizing the matrix
+        nComponents; % The block structure : dimension1 x multiplicity
+        dimensions1;
+        multiplicities;
+        types; % The representation type
+        dim; % matrix dimension
+        blocks; % The sdp blocks corresponding to each irreducible representation
+    end
+
+    methods
+        
+        function self = Centralvar(U, dimensions1, multiplicities, types)
+            try
+                yalmip('version');
+            catch
+                error('Yalmip not found in the path');
+            end
+            
+            % We set the class attributes
+            self.U = U;
+            self.nComponents = length(dimensions1);
+            self.dimensions1 = dimensions1;
+            self.multiplicities = multiplicities;
+            self.types = types;
+            self.dim = sum(self.multiplicities.*self.dimensions1);
+
+            % sanity checks
+            assert(self.nComponents == length(self.multiplicities), [num2str(self.nComponents), ' components but ', num2str(length(self.multiplicities)), ' multiplicities']);
+            assert(self.nComponents == length(self.types), [num2str(self.nComponents), ' components but ', num2str(length(self.types)), ' types']);
+            assert(self.dim == size(self.U,1), ['dimension is ', num2str(self.dim), ' but U is of size ', num2str(size(self.U,1)), 'x', num2str(size(self.U,2))]);
+            assert(self.dim == size(self.U,2), ['dimension is ', num2str(self.dim), ' but U is of size ', num2str(size(self.U,1)), 'x', num2str(size(self.U,2))]);
+
+            % We construct the SDP matrix by block
+            self.blocks = cell(1,self.nComponents);
+            for i = 1:self.nComponents
+                switch self.types(i)
```

Comment:
`replab.DivisionAlgebra` is used in this section to get for instance the matrix basis of quaternions. I don't see however how this fully answer the needs here, because blocks have to be treated differently whether they are on the diagonal or not... In what way would you make use of `replab.DivisionAlgebra` here to remove the switch?

Reply:
Looks good!